modules/telestion-application/src/main/java/de/wuespace/telestion/application/launcher/TelestionLauncher.java

```diff
@@ -0,0 +1,283 @@
+		deployments.add(deployment);
+		return deployment;
+	}
+
+	@Override
+	public VertxOptions getVertxOptions() {
+		return vertxOptions;
+	}
+
+	@Override
+	public Vertx getVertx() {
+		return vertx;
+	}
+
+	@Override
+	public Future<Void> start() throws Exception {
+		printBanner();
+		printVersion();
+		logger.debug("Start TelestionLauncher");
+
+		// parse main config
+		mainConfiguration = readMainConfig();
+		var parser = new TelestionConfigParser(mainConfiguration);
+		loaders = parser.parseLoaders();
+		vertxOptions = parser.parseVertxOptions();
+		boolean isClustered = parser.parseIsClustered();
+
+		// 1 - initialize loaders and call init event
+		return Future.future(promise -> initializeLoaders()
+				// 2 - call before vertx startup event
+				.compose(result -> Loader.call(loaders, Loader::onBeforeVertxStartup))
+				// 3 - create vertx instance
+				.compose(result -> createVertxInstance(isClustered))
+				// 4 - call after vertx startup event
+				.compose(result -> Loader.call(loaders, Loader::onAfterVertxStartup))
+				// fix future type mismatch
+				.onSuccess(result -> promise.complete())
+				.onFailure(promise::fail));
+	}
+
+	@Override
+	public Future<Void> stop() {
+		logger.debug("Stop TelestionLauncher");
+		// 1 - call before Vertx shutdown event
+		return Future.future(promise -> Loader.call(loaders, Loader::onBeforeVertxShutdown)
+				// 2 - close Vertx instance
+				.compose(result -> vertx.close())
+				// 3 - call after Vertx shutdown event
+				.compose(result -> Loader.call(loaders, Loader::onAfterVertxShutdown))
+				// 4 - clear Vertx instance from launcher
+				.compose(result -> clearVertxInstance())
+				// 5 - call exit event
+				.compose(result -> Loader.call(loaders, Loader::onExit))
+				// fix future type mismatch
+				.onSuccess(result -> promise.complete())
+				.onFailure(promise::fail));
+	}
+
+	/**
+	 * Reads the main configuration from the configured {@link #mainConfigPath}, parses it as a dynamic
+	 * {@link JsonObject} and returns it.
+	 *
+	 * @return the parsed main configuration as a dynamic {@link JsonObject}
+	 */
+	private JsonObject readMainConfig() throws IOException {
+		logger.debug("Read main configuration from {}", mainConfigPath);
+		return new JsonObject(Files.readString(mainConfigPath));
+	}
+
+	/**
+	 * Initializes a registered {@link Loader Loaders} in this {@link TelestionLauncher} instance.
+	 * <p>
+	 * It returns a {@link Future} which resolves when all loaders are initialized.
+	 *
+	 * @return a future which resolves on successful loader initialization
+	 */
+	private Future<CompositeFuture> initializeLoaders() {
+		logger.debug("Initialize loaders");
+		loaders.forEach(loader -> loader.init(this));
+		return Loader.call(loaders, Loader::onInit);
+	}
+
+	/**
+	 * Creates a new {@link Vertx} instance with the {@link #vertxOptions} from the {@link TelestionLauncher} instance.
+	 * <p>
+	 * A future returns the instance when it resolves.
+	 *
```

Comment:
This line makes doesn't work grammatically, but it looks like it's supposed to describe the return value => that's described below, anyways

```suggestion
	 * <p>
	 *
```

Reply:
Then we can remove the `<p>` tag, too :wink:

Reply:
Done in [`94f4808` (#504)](https://github.com/wuespace/telestion-core/pull/504/commits/94f48089b7c8f198c1c3b37f7b41d800e9686f58)